dev/com.ibm.ws.artifact.zip/src/com/ibm/ws/artifact/zip/internal/JavaArchive.java

```diff
@@ -0,0 +1,575 @@
+		}
+		
+		long offsetFromBeginningOfFile = zip64EoCDOffset;
+		int offsetInBlock = 0;
+		int offsetFromStartingPoint = 0;   // The starting point is zip64EoCDOffset;
+
+		// We will only enter this loop if data has been attached to the front of the archive (or if
+		// the archive is invalid.)  The loop searches forward in the file starting at the point where
+		// the Zip64 Locator indicates the Zip64-End-of-Central-Directory-Record should be.
+		while ( !isValid_Zip64EoCDRecord( block, offsetInBlock, offsetFromBeginningOfFile ) ) {
+			offsetInBlock++;
+			offsetFromBeginningOfFile++;
+			offsetFromStartingPoint++;
+			
+			// An artificial limit is placed on the size of data allowed to be prepended to an archive.
+			// This is because this implementation searches through the file.  A fruitless search through
+			// a large file would cause an unacceptable delay if we did not set a limit.
+			if (offsetFromStartingPoint > MAX_PREFIX_ALLOWED) {
+			     throw new IOException("Failed to find the Zip64-End-of-Central-Directory-Record after searching " + MAX_PREFIX_ALLOWED + " bytes.");	
+			}
+			
+			if ( offsetInBlock > block.length - 4 ) {  // Must allow 4 bytes at end of block to read signature
+
+				block = getBlockFromBeginningOfFile(file, offsetFromBeginningOfFile, FORWARD_BLOCK_SIZE);
+				if (block.length < 4) {
+					throw new IOException("Reached end of file while searching for to Zip64-End-of-Central-Directory-Record does not correct.");
+				}
+				offsetInBlock = 0;
+			}
+		}
+
+		if (block.length < (offsetInBlock + ZIP64_EOCDR_MINIMUM_SIZE ))  {
+			block = getBlockFromBeginningOfFile(file, offsetFromBeginningOfFile, ZIP64_EOCDR_MINIMUM_SIZE);
+		    return  Arrays.copyOfRange(block, 0, ZIP64_EOCDR_MINIMUM_SIZE);
+		} 
+		
+		// Only store the minimum size for this record.  We don't need to store any extra data.
+		return Arrays.copyOfRange(block, offsetInBlock , offsetInBlock + ZIP64_EOCDR_MINIMUM_SIZE);
+	}
+	
+	/**
+	 * @param file a zip file
+	 * @param size number of bytes to read from end of file.
+	 * @return byte[] an array of data of 'size' bytes from the end of the file
+	 * @throws IOException
+	 */
+	private byte[] getBlockFromEndOfFile(RandomAccessFile file, int size) throws IOException {
+
+		int blockLength =  size >_fileLength ? (int)_fileLength : size;
+		long offset = _fileLength - blockLength;
+
+		file.seek(offset);
+
+		byte[] bytes = new byte[blockLength];
+		file.read(bytes, 0, blockLength );
+		return bytes;
+	}
+	
+	/**
+	 * 
+	 * @param file  a zip file
+	 * @param offsetFromEnd Offset from end of file where reading begins
+	 * @param size number of bytes of data to read
+	 * @return Returns an array of 'size' bytes that starts from 'offsetFromEnd' of file.
+	 * @throws IOException
+	 */
+	private byte[] getBlockFromEndOfFile(RandomAccessFile file, int offsetFromEnd, int size) throws IOException {
+		
```

Comment:
For better performance, the block can be allocated outside of the method, and reused when there are several calls.  But, since this code is a very rare case, that level of optimization doesn't seem necessary.

Reply:
done